sdk/lib/opentelemetry/sdk/trace/export/batch_span_processor.rb

```diff
@@ -136,12 +136,13 @@ def force_flush(timeout: nil) # rubocop:disable Metrics/AbcSize, Metrics/Cycloma
           #   non-specific failure occurred, TIMEOUT if a timeout occurred.
           def shutdown(timeout: nil)
             start_time = Time.now
-            lock do
+            thread_alive = lock do
               @keep_running = false
               @condition.signal
+              !@thread.nil? && Process.pid == @pid
             end
 
-            @thread.join(timeout)
+            @thread.join(timeout) if thread_alive
```

Comment:
Can we clear `@thread` in `reset_on_fork` if `restart_thread == false`? Then:
```ruby
thread = lock do
  @keep_running = false
  @condition.signal
  @thread
end

thread&.join(timeout)
```

Gating the validity of `@thread` on `Process.pid == @pid` seems error prone (for future changes to this code).

Reply:
Sounds good. Done. (This requires, of course, that we also add a call to `reset_on_fork(restart_thread: false)` so that `@thread` actually does get cleared to nil.)

Reply:
> This requires, of course, that we also add a call to `reset_on_fork(restart_thread: false)` so that `@thread` actually does get cleared to nil.

Does it? It shouldn't be necessary because the `@thread` state should correctly reflect whether we're in a process with the worker thread running.

Reply:
Hmm, you're right. Looks like you can safely call `Thread#join` even in the child process. The thread object is non-nil but treats its state as a "dead" (i.e. terminated) thread.

Okay, I removed the `reset_on_fork` call, and removed the test for that case since it's a non-issue.